include/seqan/platform.h

```diff
@@ -401,7 +401,11 @@ typedef int8_t __int8;     // nolint
 
 // A macro to eliminate warnings on GCC and Clang
 #if defined(COMPILER_GCC) || defined(COMPILER_CLANG) || defined(COMPILER_LINTEL)
-#define SEQAN_UNUSED __attribute__((unused))
+#if __cplusplus < 201703
+#define SEQAN_UNUSED [[gnu::unused]]
+#else
+#define SEQAN_UNUSED [[maybe_unused]]
+#endif // __cplusplus < 201703
 #else
 #define SEQAN_UNUSED
 #endif
```

Comment:
The if clauses are wrong. because the macro is never expanded on MSVC. You need to check for ` __cplusplus < 201703` first and then if it is not the case check for the compilers and if possible to define the gnu attribute.

Reply:
Good catch. 🔍